bin/q

```diff
@@ -43,13 +43,17 @@ import codecs
 import locale
 import time
 import re
-from ConfigParser import ConfigParser
+from six.moves import configparser, range, filter
 import traceback
 import csv
 import hashlib
 import uuid
-import cStringIO
 import math
+import six
+
+if six.PY3:
+    long = int
+    unicode = six.text_type
```

Comment:
This approach does not follow https://docs.python.org/3/howto/pyporting.html#use-feature-detection-instead-of-version-detection  Also __unicode(x, encoding)__ will fail in Python 3 and you have an instance of that in this file.

Reply:
removed the unicode(x, encoding) code, thanks for noticing. As for the long/unicode checking, do you happen to have a suggestion on how to do that using feature-detection?